File: src/main/java/com/footverse/category/dto/CategoryResponse.java
package com.footverse.category.dto;

/**
 * A product category returned to clients (dto-spec §10).
 *
 * @param id          the category id
 * @param name        the unique category name
 * @param description the description, if any
 */
public record CategoryResponse(
        Long id,
        String name,
        String description) {
}

File: app/lib/features/product/models/category_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'category_response.g.dart';

/// A product category (dto-spec §10).
@JsonSerializable(createToJson: false)
class CategoryResponse {
  const CategoryResponse({
    required this.id,
    required this.name,
    this.description,
  });

  factory CategoryResponse.fromJson(Map<String, dynamic> json) =>
      _$CategoryResponseFromJson(json);

  final int id;
  final String name;
  final String? description;
}
